In [3]:
import pandas as pd
import matplotlib.pyplot as plt

In [4]:
df = pd.read_csv("updated_dataset.csv")
df.head()

distance  surge_multiplier  price  Black  Black SUV  Lux  Lux Black  \
0      0.44               1.0    5.0      0          0    0          0   
1      0.44               1.0   11.0      0          0    1          0   
2      0.44               1.0    7.0      0          0    0          0   
3      0.44               1.0   26.0      0          0    0          0   
4      0.44               1.0    9.0      0          0    0          0   

   Lux Black XL  Lyft  Lyft XL  Shared  UberPool  UberX  UberXL  WAV  Lyft.1  \
0             0     0        0       1         0      0       0    0       1   
1             0     0        0       0         0      0       0    0       1   
2             0     1        0       0         0      0       0    0       1   
3             1     0        0       0         0      0       0    0       1   
4             0     0        1       0         0      0       0    0       1   

   Uber  
0     0  
1     0  
2     0  
3     0  
4     0

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 632387 entries, 0 to 632386
Data columns (total 17 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   distance          632387 non-null  float64
 1   surge_multiplier  632387 non-null  float64
 2   price             632387 non-null  float64
 3   Black             632387 non-null  int64  
 4   Black SUV         632387 non-null  int64  
 5   Lux               632387 non-null  int64  
 6   Lux Black         632387 non-null  int64  
 7   Lux Black XL      632387 non-null  int64  
 8   Lyft              632387 non-null  int64  
 9   Lyft XL           632387 non-null  int64  
 10  Shared            632387 non-null  int64  
 11  UberPool          632387 non-null  int64  
 12  UberX             632387 non-null  int64  
 13  UberXL            632387 non-null  int64  
 14  WAV               632387 non-null  int64  
 15  Lyft.1            632387 non-null  int64  
 16  Uber              63

In [6]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression


In [7]:
model = LinearRegression()
x = df.drop('price',axis=1)
y = df['price']
print(x.shape,y.shape)

(632387, 16) (632387,)


In [8]:
xtrain,xtest,ytrain,ytest = train_test_split(x,y,test_size=0.2,random_state=42015)
print(xtrain.shape,ytrain.shape,xtest.shape,ytest.shape)

(505909, 16) (505909,) (126478, 16) (126478,)


In [9]:
model.fit(xtrain,ytrain)
ypred = model.predict(xtest)

In [10]:
from sklearn.metrics import r2_score,mean_absolute_percentage_error
score = r2_score(ytest,ypred)
print(score)

0.9338125110563765


In [11]:
mape = mean_absolute_percentage_error(ytest,ypred)
mape

0.12883434302166052

The Coefficient of Determination (R2) is 0.9338

Mean Abs Percent Error == 12.88%